packages/dashboard-core-plugins/src/ConsolePlugin.tsx

```diff
@@ -547,26 +548,11 @@ export function ConsolePlugin(
     [notebooksUrl]
   );
 
-  useEffect(
-    function registerComponentsAndReturnCleanup() {
-      const cleanups = [
-        registerComponent(ConsolePanel.COMPONENT, ConsolePanel, hydrateConsole),
-        registerComponent(CommandHistoryPanel.COMPONENT, CommandHistoryPanel),
-        registerComponent(FileExplorerPanel.COMPONENT, FileExplorerPanel),
-        registerComponent(LogPanel.COMPONENT, LogPanel),
-        registerComponent(
-          NotebookPanel.COMPONENT,
-          NotebookPanel,
-          hydrateNotebook
-        ),
-      ];
-
-      return () => {
-        cleanups.forEach(cleanup => cleanup());
-      };
-    },
-    [registerComponent, hydrateConsole, hydrateNotebook]
-  );
+  usePanelRegistration(registerComponent, ConsolePanel, hydrateConsole);
```

Comment:
Yea it could be, along with everything else that's passed in via `cloneElement` in [DashboardLayout](https://github.com/deephaven/web-client-ui/blob/3247d274dd724c5c65caa7047ad843c03d42aee4/packages/dashboard/src/DashboardLayout.tsx#L320) (`panelManager`, `layout`, `id`)